packages/leaflet-map/src/base-map.tsx

```diff
@@ -0,0 +1,262 @@
+import { useState, useEffect, PropsWithChildren } from 'react';
+import {loadWidget} from '../../lib/load-widget';
+import { LatLng, latLngBounds, LeafletMouseEvent } from 'leaflet';
+import { MapContainer, useMapEvents } from 'react-leaflet';
+import { MapConsumer, useMapRef } from './map-consumer';
+import TileLayer from './tile-layer';
+import { Area, isPointInArea } from './area';
+import Marker from './marker';
+import MarkerClusterGroup from './marker-cluster-group';
+
+// ToDo: import { searchAddressByLatLng, suggestAddresses, LookupLatLngByAddressId } from './lib/search.js';
+
+import 'leaflet/dist/leaflet.css';
+import './css/base-map.less';
+
+import { LocationType } from './types/location';
+import { BaseProps } from '../../types/base-props';
+import { ProjectSettingProps } from '../../types/project-setting-props';
+import { MarkerProps } from './types/marker-props';
+import { MapPropsType } from './types/index';
+
+export type BaseMapWidgetProps =
+  BaseProps &
+  ProjectSettingProps & {
+    resourceId?: string;
+  } &
+  MapPropsType;
+
+export function BaseMap({
+
+  iconCreateFunction = undefined,
+  defaultIcon = undefined,
+
+	area = undefined,
+  areaPolygonStyle = undefined,
+
+  markers = [],
+
+  autoZoomAndCenter = undefined,
+
+	zoom = 14,
+  scrollWheelZoom = true,
+	center = {
+		lat: 52.37104644463586,
+		lng: 4.900402911007405,
+	},
+
+  tilesVariant = 'nlmaps',
+  tiles = undefined,
+  minZoom = 0,
+  maxZoom = 25,
+
+  categorize = undefined,
+  
+  // ToDo: search = false,
+	zoomposition = 'bottomleft',
+	disableDefaultUI = true,
+  clustering = {
+    isActive: true,
+  },
+
+  onClick = undefined,
+  onMarkerClick = undefined,
+
+  ...props
+}: PropsWithChildren<BaseMapWidgetProps>) {
+
+  let [, setMapIsReady] = useState(false);
+  let [currentMarkers, setCurrentMarkers] = useState(markers);
+
+  let [mapId] = useState(`${parseInt(Math.random() * 1e8 as any as string)}`);
+  let [mapRef] = useMapRef(mapId);
+
+  // map is ready
+  useEffect(() => {
+		setMapIsReady(true);
+		let event = new CustomEvent('osc-map-is-ready', { detail: { id: mapId } });
+		window.dispatchEvent(event);
+  }, []);
+
+	// auto zoom and center on init
+  useEffect(() => {
+    if (!mapRef) return;
+	  if (autoZoomAndCenter) {
+		  let centerOn = ( autoZoomAndCenter == 'area' && area ) || ( currentMarkers && currentMarkers.length && currentMarkers );
```

Comment:
currentMarkers staat twee keer in dezelfde if statement
( currentMarkers && currentMarkers.length && currentMarkers )

Reply:
Dat klopt: het resultaat van die vergelijking, dus de inhoud van centerOn, is zo de inhoud van currentMarkers. Zonder die tweede is het resultaat de lengte van de array. Kwa leesbaarheid indrerdaad niet optimaal.

Reply:
```
currentMarkers?.length && currentMarkers
```
is denk ik iets beter

Reply:
Het zijn nu ifs geworden